lib/ansible/modules/network/ovs/openvswitch_bridge.py

```diff
@@ -109,276 +112,173 @@
+                               " %(bridge)s")
+        command = templatized_command % module.params
+        rc, out, err = module.run_command(command, check_rc=True)
+        obj['fail_mode'] = out.strip()
+
+        templatized_command = ("%(ovs-vsctl)s -t %(timeout)s br-get-external-id"
+                               " %(bridge)s")
+        command = templatized_command % module.params
+        rc, out, err = module.run_command(command, check_rc=True)
+        obj['external_ids'] = _external_ids_to_dict(out)
+
+    return obj
+
+
+def map_params_to_obj(module):
+    obj = {
+        'bridge': module.params['bridge'],
+        'parent': module.params['parent'],
+        'vlan': module.params['vlan'],
+        'fail_mode': module.params['fail_mode'],
+        'external_ids': module.params['external_ids'],
+        'set': module.params['set']
+    }
+
+    return obj
 
 # pylint: disable=E0602
 def main():
     """ Entry point. """
-    module = AnsibleModule(
-        argument_spec={
-            'bridge': {'required': True},
-            'parent': {'default': None},
-            'vlan': {'default': None, 'type': 'int'},
-            'state': {'default': 'present', 'choices': ['present', 'absent']},
-            'timeout': {'default': 5, 'type': 'int'},
-            'external_ids': {'default': None, 'type': 'dict'},
-            'fail_mode': {'default': None},
-            'set': {'required': False, 'default': None}
-        },
-        supports_check_mode=True,
-    )
-
-    bridge = OVSBridge(module)
-    if module.check_mode:
-        bridge.check()
-    else:
-        bridge.run()
+    argument_spec={
+        'bridge': {'required': True},
+        'parent': {'default': None},
+        'vlan': {'default': None, 'type': 'int'},
+        'state': {'default': 'present', 'choices': ['present', 'absent']},
+        'timeout': {'default': 5, 'type': 'int'},
+        'external_ids': {'default': None, 'type': 'dict'},
+        'fail_mode': {'default': None},
+        'set': {'required': False, 'default': None}
+    }
+
+    required_if = [('parent', not None, ('vlan',))]
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           required_if=required_if,
+                           supports_check_mode=True)
+
+    result = {'changed': False}
+
+    # We add ovs-vsctl to module_params to later build up templatized commands
+    module.params["ovs-vsctl"] = module.get_bin_path("ovs-vsctl", True)
+
+    want = map_params_to_obj(module)
+    have = map_config_to_obj(module)
+
+    commands = map_obj_to_commands(want, have, module)
+    result['commands'] = commands
+
+    if commands:
+        if not module.check_mode:
+            for c in commands:
+                module.run_command(c, check_rc=True)
+        result['changed'] = True
+
+    module.exit_json(**result)
+
 
 # pylint: disable=W0614
 # pylint: disable=W0401
 # pylint: disable=W0622
 
 # import module snippets
 from ansible.module_utils.basic import *
```

Comment:
Should probably move these imports up and lose the * import, that hasn't been necessary since 2.2(?)